hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -129,8 +135,10 @@ loadChainContext config party = do
 mkTinyWallet ::
   Tracer IO DirectChainLog ->
   DirectChainConfig ->
+  -- | The current running era we can use to query the node
+  CardanoEra era ->
   IO (TinyWallet IO)
-mkTinyWallet tracer config = do
+mkTinyWallet tracer config era = do
```

Comment:
This is an `era` argument we should be able to get rid as well (it impacts quite some tests)

In detail, this means we would need to make `queryProtocolParameters` also query the current era on the same connection. It might be a bit tricky to witness the right `era` types in there, but that would be a good next step.

Reply:
nice catch!